plugins/ledger/config/initializers/ledger/client.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+#
+# basic ledger event client
+#
+require 'faraday'
+require 'openssl'
+
+LEDGER_URI = "/api/v1/events"
+
+module Ledger
+  class Client
+    class << self
+      def verify_env
+        raise "LEDGER_TOKEN is not set in ENV" unless ENV.fetch("LEDGER_TOKEN", false)
+        raise "LEDGER_BASE_URL is not set in ENV" unless ENV.fetch("LEDGER_BASE_URL", false)
+      end
+
+      def post_deployment(deploy)
+        verify_env
+        deployment(deploy)
+      end
+
+      private
+
+      def create_event(event)
+        events = { "events": [event] }
+        results = post(events)
+        # TODO: raise or display errors if results.status != 200
+        results
+      end
+
+      def deployment(deploy)
+        event = {
+          id:           deploy.id,
+          name:         deploy.project.name,
+          actor:        deploy.user.name,
+          status:       deploy.status,
+          started_at:   deploy.created_at.iso8601,
+          summary:      deploy.summary,
+          environment:  deploy.stage.deploy_groups.map(&:environment).uniq.map(&:name).join(","),
```

Comment:
might want to use permalink here and/or downcase to avoid `Production` vs `production` ... possibly also not do anything if env name is not in the list of `master, staging, production` since that would just confuse ledger